app/src/main/java/com/doubtless/doubtless/main/bottomNav/RetroBottomNav.kt

```diff
@@ -0,0 +1,127 @@
+package com.doubtless.doubtless.main.bottomNav
+
+import android.content.Context
+import android.os.Bundle
+import android.os.Parcelable
+import android.os.Vibrator
+import android.os.VibratorManager
+import android.util.AttributeSet
+import android.view.View
+import android.widget.LinearLayout
+import androidx.core.view.children
+import com.doubtless.doubtless.theming.buttons.SecondaryButton
+
+class RetroBottomNav(
+    context: Context,
+    attributeSet: AttributeSet
+) : LinearLayout(context, attributeSet) {
+
+    private var currentSelectedIndex: Int? = null
+    private val elements: ArrayList<BottomIntractableElement> = arrayListOf()
+    private var onSelectedItemChangedListener: OnSelectedItemChangedListener? = null
+
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        elements.addAll(findBottomNavElements())
+
+        // set initial selected state
+        if (currentSelectedIndex == null) { // not restored after process death.
+            currentSelectedIndex = BottomNavMenu.defaultSelectedIndex
+        }
+
+        onNewSelectedIndex()
+        notifyAllUnSelectedElements()
+
+        // set click listeners on each button
+        elements.forEachIndexed { idx, btn ->
+            (btn as View).setOnClickListener {
+                setSelectStateOfSelectedItemAndNotifyUnselectedForElse(idx, btn)
+            }
+        }
+    }
+
+    private fun onNewSelectedIndex() {
+        onSelectedItemChangedListener?.onNewSelectedIndex(currentSelectedIndex!!)
+
+        elements[currentSelectedIndex!!].onSelected()
```

Comment:
I think rather than using ‘!!’ Operator, we should use default index if currIdx is null. Even though it wont be null anytime, but using a safer approach would be better

Reply:
Ok will do. Though I prefer crashing the app rather than silently consuming the bug, most users don't know what a crash is and think that the app closed abruptly due to some random reason and will ignore it. but thats a personal opinion.